pkg/gatherers/conditional/gather_logs_of_namespace.go

```diff
@@ -0,0 +1,88 @@
+package conditional
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openshift/insights-operator/pkg/gatherers"
+	"github.com/openshift/insights-operator/pkg/gatherers/common"
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// BuildGatherLogsOfNamespace creates a gathering closure which collects logs from pods in the provided namespace
+// Params:
+//   - namespace string - namespace from which to collect image streams
+//   - label_selector string - a label selector to filter some pods (default to all pods)
+//   - tail_lines int64 - a number of log lines to keep for each container
+//
+// The Kubernetes API:
+//          https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/pod_expansion.go#L48
+// Response see:
+//          https://docs.openshift.com/container-platform/4.6/rest_api/workloads_apis/pod-core-v1.html#apiv1namespacesnamespacepodsnamelog
+//
+// * Location in archive: conditional/namespaces/{namespace}/pods/{pod_name}/containers/{container_name}/logs/last-{n}-lines.log
```

Comment:
OK so we shouldn't forget about our generated docs. Can you please run `make docs`? But this will not be generated anyway. Does it make sense to start the name of the method with `Build` (the gendoc scans only methods starting with `Gather` AFAIK)? 
We can also add here:
```
* Since versions:
//   * 4.8+
``` 
Same for the above `gather_image_stream_definitions_of_namespace.go` gatherer.

Reply:
I called it `BuildX` because these are the types of gatherers that can have parameters and they are a little different than normal ones. Lemme check the docs generator.